### PR TITLE
Remove Ready to Run from Pool

### PR DESCRIPTION
**Pull request by callunity**
GLT-1891

### CODE REVIEW

**Comment by sqwci** [comment]
SonarQube analysis reported 4 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![CRITICAL][CRITICAL] [Dtos.java#L432](https://github.com/TGAC/miso-lims/blob/c304e240f5184026a0fae9c25b0583fd602a05e1/miso-dto/src/main/java/uk/ac/bbsrc/tgac/miso/dto/Dtos.java#L432): Refactor this method to reduce its Cognitive Complexity from 24 to the 15 allowed. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3776)
1. ![CRITICAL][CRITICAL] [EditLibraryController.java#L649](https://github.com/TGAC/miso-lims/blob/c304e240f5184026a0fae9c25b0583fd602a05e1/miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/EditLibraryController.java#L649): Define a constant instead of duplicating this literal "sortableLocation" 3 times. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1192)
1. ![MAJOR][MAJOR] [PoolRestController.java#L164](https://github.com/TGAC/miso-lims/blob/c304e240f5184026a0fae9c25b0583fd602a05e1/miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/rest/PoolRestController.java#L164): Refactor the code so this stream pipeline is used. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3958)
1. ![MINOR][MINOR] [PoolRestController.java#L280](https://github.com/TGAC/miso-lims/blob/c304e240f5184026a0fae9c25b0583fd602a05e1/miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/rest/PoolRestController.java#L280): Replace this lambda with a method reference. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1612)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/EditLibraryController.java [hunk 1]
```diff
@@ -150,6 +150,16 @@
     INDEX_FAMILY_NEEDS_PLATFORM.setName("Please select a platform...");
   }
 
+  private static class Config {
```
Comment (sqwci) [inline miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/EditLibraryController.java]: MAJOR Add a private constructor to hide the implicit public one. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1118)

File: miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/EditSampleController.java [hunk 2]
```diff
@@ -253,6 +253,17 @@ public String getDefaultSciName() {
     return defaultSciName != null ? defaultSciName : "";
   }
 
+  private static class Config {
```
Comment (sqwci) [inline miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/EditSampleController.java]: MAJOR Add a private constructor to hide the implicit public one. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1118)

**Comment by sqwci** [comment]
SonarQube analysis reported 5 issues
* ![CRITICAL][CRITICAL] 1 critical
* ![MAJOR][MAJOR] 3 major
* ![MINOR][MINOR] 1 minor

Watch the comments in this conversation to review them.

#### 3 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![CRITICAL][CRITICAL] [EditSampleController.java#L755](https://github.com/TGAC/miso-lims/blob/7d8f03a80c70478180e063dafed2acacac5a8198/miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/EditSampleController.java#L755): Refactor this method to reduce its Cognitive Complexity from 18 to the 15 allowed. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3776)
1. ![MAJOR][MAJOR] [PoolRestController.java#L164](https://github.com/TGAC/miso-lims/blob/7d8f03a80c70478180e063dafed2acacac5a8198/miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/rest/PoolRestController.java#L164): Refactor the code so this stream pipeline is used. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3958)
1. ![MINOR][MINOR] [PoolRestController.java#L280](https://github.com/TGAC/miso-lims/blob/7d8f03a80c70478180e063dafed2acacac5a8198/miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/rest/PoolRestController.java#L280): Replace this lambda with a method reference. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1612)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: miso-dto/src/main/java/uk/ac/bbsrc/tgac/miso/dto/Dtos.java [hunk 3]
```diff
@@ -133,6 +133,7 @@
 import uk.ac.bbsrc.tgac.miso.core.util.BoxUtils;
 import uk.ac.bbsrc.tgac.miso.core.util.LimsUtils;
 
+@SuppressWarnings("squid:S3776")
```
Comment (djcooke) [inline miso-dto/src/main/java/uk/ac/bbsrc/tgac/miso/dto/Dtos.java]: Could you add a comment to translate the number?

File: miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/EditSampleController.java [hunk 4]
```diff
@@ -139,6 +139,7 @@
 import uk.ac.bbsrc.tgac.miso.webapp.util.BulkEditTableBackend;
 import uk.ac.bbsrc.tgac.miso.webapp.util.BulkPropagateTableBackend;
 
+@SuppressWarnings("squid:S3776")
```
Comment (djcooke) [inline miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/EditSampleController.java]: Would rather not have this here, as a couple of the methods are pretty ugly and it's good that Sonar complains about it

**Comment by sqwci** [comment]
SonarQube analysis reported 4 issues
* ![MAJOR][MAJOR] 3 major
* ![MINOR][MINOR] 1 minor

Watch the comments in this conversation to review them.

#### 2 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [PoolRestController.java#L164](https://github.com/TGAC/miso-lims/blob/050427bb02524651b583c6d5b0d3cef604044415/miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/rest/PoolRestController.java#L164): Refactor the code so this stream pipeline is used. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3958)
1. ![MINOR][MINOR] [PoolRestController.java#L280](https://github.com/TGAC/miso-lims/blob/050427bb02524651b583c6d5b0d3cef604044415/miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/rest/PoolRestController.java#L280): Replace this lambda with a method reference. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1612)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by sqwci** [comment]
SonarQube analysis reported 4 issues
* ![MAJOR][MAJOR] 3 major
* ![MINOR][MINOR] 1 minor

Watch the comments in this conversation to review them.

#### 2 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [PoolRestController.java#L164](https://github.com/TGAC/miso-lims/blob/d5efc1558bd5efc2adb23f4d81a6d1d22827d63a/miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/rest/PoolRestController.java#L164): Refactor the code so this stream pipeline is used. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3958)
1. ![MINOR][MINOR] [PoolRestController.java#L280](https://github.com/TGAC/miso-lims/blob/d5efc1558bd5efc2adb23f4d81a6d1d22827d63a/miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/rest/PoolRestController.java#L280): Replace this lambda with a method reference. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1612)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Pull request merged**